### PR TITLE
feat: Naive finetuning

### PR DESCRIPTION
**Pull request by lucapericlp**
Taking the approach of freezing the entire LLM other than the `[-last_n_blocks_to_finetune:]` & the last norm layer. Using the "default" hyperparams, we're seeing ~12Gb VRAM usage.

### CODE REVIEW

File: datasets/sample_dataset.csv [hunk 1]
```diff
@@ -0,0 +1,321 @@
+audio_files|captions
```
Comment (lucapericlp) [inline datasets/sample_dataset.csv:1]: Opted to omit from `git lfs` to avoid setup complexity

**Review by vatsalaggarwal — DISMISSED** [review]
nice!

File: fam/llm/config/finetune_params.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        raise Exception("out_dir should be just a name, not a path with slashes")
+
+    ckpts_save_dir = ckpts_base_dir / out_dir
+    os.makedirs(ckpts_save_dir, exist_ok=True)
+
+causal = True
+bias: bool = False  # do we use bias inside LayerNorm and Linear layers?
+spk_emb_on_text: bool = True  # whether to add speaker embedding conditioning to text tokens or not
+
+torch.manual_seed(seed + seed_offset)
```
[the remark this answers is not in the document]
Comment (lucapericlp) [reply fam/llm/config/finetune_params.py:91]: Yeah makes sense!

File: fam/llm/preprocessing/audio_token_mode.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    num_hierarchies = audio_tokens.shape[0]
+    assert num_hierarchies >= 2, f"Unexpected number of hierarchies: {num_hierarchies}. Expected at least 2."
+
+    # choosing -5 so that we can't get error!
+    interleaved_audio_tokens = np.full((len(audio_tokens[0]) + len(audio_tokens[1]),), -5)
+    interleaved_audio_tokens[::2] = audio_tokens[0]
+    interleaved_audio_tokens[1::2] = audio_tokens[1] + second_hierarchy_flattening_offset
+
+    # TODO: check that text_tokens can get different offset?
```
[the remark this answers is not in the document]
Comment (lucapericlp) [reply fam/llm/preprocessing/audio_token_mode.py:30]: Good catch 👍

File: fam/llm/config/finetune_params.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+speaker_cond = True
+
+# always running finetuning on a single GPU
+master_process = True
+device: str = "cuda"  # examples: 'cpu', 'cuda', 'cuda:0', 'cuda:1' etc., or try 'mps' on macbooks
+ddp = False
+ddp_world_size = 1
+tokens_per_iter = gradient_accumulation_steps * ddp_world_size * batch_size * block_size
+
+print(f"tokens per iteration will be: {tokens_per_iter:,}")
```
Comment (vatsalaggarwal) [inline fam/llm/config/finetune_params.py:70]: nit: move all the code below into train.py

**Review by vatsalaggarwal — APPROVED** [review]
small nit, good to go!

**Review by vatsalaggarwal — APPROVED** [review]

**Pull request merged**